### PR TITLE
cam6_3_110: Resolve miscellaneous issues.

### LINKED ISSUE #371: Grid matches in refcases will need to change from gland4 to gris4 if they are re-enabled
CAM has a bunch of currently-disabled refcases:

https://github.com/ESCOMP/CAM/blob/e0810cd1240fc9a165b15152554cd87390299282/cime_config/config_compsets.xml#L600-L641

If they are re-enabled, the grid match will need to change from gland4 to gris4, since that is the new name for the Greenland grid, once https://github.com/ESMCI/cime/pull/3943 comes in (tentatively scheduled for cesm2_3_alpha04b).

### LINKED ISSUE #608: Fix path for buildlib/buildnml
### What happened?

Accroding to Erik Kluzek:

Change path for components buildlib/buildnml from cime/scripts/Tools to cime/CIME/Tools. We had some discussion over email. The previous path no longer exists, so it's at least confusing to set a path that no longer exists. But, it also might break in the future if it's not corrected. And having the right path also allows you to run these scripts outside on their own, which is sometimes useful.

### What are the steps to reproduce the bug?

N/A

### What CAM tag were you using?

cam6_3_060

### What machine were you running CAM on?

Other (please explain below)

### What compiler were you using?

Other (please specify below)

### Path to a case directory, if applicable

_No response_

### Will you be addressing this bug yourself?

No

### Extra info

_No response_

### LINKED ISSUE #741: Remove 16 character limitation for CLUBB variables being output
### Issue Type

Code Clean-up

### Issue Description

A 16 character limitation was introduced early on in CLUBB due to a limitation in addfld and the length of max_fieldname_len being 16.  When this variable was increased in size, the restriction in CLUBB was not removed.  

All limitations on CLUBB variable lengths can now be removed as the current length appears to be 35 characters which should be adequate for all CLUBB variables.

### Will this change answers?

No

### Will you be implementing this yourself?

Any CAM SE can do this

### LINKED ISSUE #749: Correct description of low-level wind output
### What is the feature/what would you like to discuss?

In `cam_diagnostics.F90` the output variables `WSPDSRFMX` and `WSPDSRFAV` are named incorrectly in subroutine `diag_init_dry`:

```
call addfld ('WSPDSRFMX',  horiz_only,  'X', 'm/s',       'Horizontal total wind speed maximum at the surface' )
call addfld ('WSPDSRFAV',  horiz_only,  'A', 'm/s',       'Horizontal total wind speed average at the surface' )
```

They're not actually "at the surface", they are "in the lowest model level" as is clear when they are written to output in subroutine `diag_phys_writeout_dry`:

```
! Wind speed
ftem(:ncol,:) = sqrt( state%u(:ncol,:)**2 + state%v(:ncol,:)**2)
call outfld ('WSPEED  ',ftem    ,pcols   ,lchnk     )
call outfld ('WSPDSRFMX',ftem(:,pver)   ,pcols   ,lchnk     )
call outfld ('WSPDSRFAV',ftem(:,pver)   ,pcols   ,lchnk     )
```

This is *very* likely to be misinterpreted by users as being comparable to 10m wind in observations. 

I think all that needs to be changed is the description string. I'd suggest:

```
call addfld ('WSPDSRFMX',  horiz_only,  'X', 'm/s', 'Horizontal total wind speed maximum at the lowest model level' )
call addfld ('WSPDSRFAV',  horiz_only,  'A', 'm/s',   'Horizontal total wind speed average at the lowest model level' )
```

### Is there anyone in particular you want to be part of this conversation?

_No response_

### Will this change (regression test) answers?

No

### Will you be implementing this enhancement yourself?

No

### LINKED ISSUE #803: Add missing GHGMAM4 to FMTHIST compset
### What happened?

FMTHIST does not have %GHGMAM4 qualifier.  

Bump the compset designators from "v0b" to "v0c" for both FMTHIST and FLTHIST

### What are the steps to reproduce the bug?

none

### What CAM tag were you using?

cam6_3_109

### What machine were you running CAM on?

CISL machine (e.g. cheyenne)

### What compiler were you using?

Intel

### Path to a case directory, if applicable

_No response_

### Will you be addressing this bug yourself?

Any CAM SE can do this

### Extra info

See the discussion at: https://github.com/NCAR/amwg_dev/issues/260

### PR DESCRIPTION
**Pull request by brian-eaton**
Resolve issues:

. Issue #371 - Grid matches in refcases will need to change from gland4 to gris4 if they are re-enabled
    - Resolved by removing entries for old refcases.

. Issue #741 - Updates to address CLUBB variable-name length limit.
    - Use cam_history_support::max_fieldname_len to replace hardcoded 16.
    - PR #743 will be resolved by this PR.

. Issue #608 - Fix path for buildlib/buildnml
    - Fix one wrong occurance of the old path 'cime/scripts/Tools'.  Don't see any problem with paths for buildlib/buildnml.

. Issue #749 - Correct description of low-level wind output
    - fix description as suggested

. resolves #371
. resolves #741
. resolves #743
. resolves #608
. resolves #749
. resolves #803

### CODE REVIEW

File: doc/ChangeLog [hunk 1]
```diff
@@ -1,5 +1,120 @@
 ===============================================================
 
+Tag name:
+Originator(s): eaton
```
Comment (cacraigucar) [inline doc/ChangeLog:4]: Might want to include bstephens as well since you are bringing in his PR.

**Review by cacraigucar — APPROVED** [review]

**Review by fvitt — APPROVED** [review]
Seems OK to me.

File: doc/ChangeLog [hunk 2]
```diff
@@ -1,5 +1,120 @@
...
+
+src/control/cam_history.F90
+. remove code from subroutine fldlst that checks for names of zonal mean
+  fields in the fincl lists which were only available when the FV dycore
+  was used.  Since new functionality has been implemented to make these
+  zonal mean diagnostic fields available for all dycores, the log file
+  messages being issued are incorrect.
+
+src/physics/cam/cam_diagnostics.F90
+. change descriptions of WSPDSRFMX and WSPDSRFAV for 'at the surface' to
```
Comment (nusbaume) [inline doc/ChangeLog:68]: Typo here (`for` -> `from`).

**Review by nusbaume — APPROVED** [review]
Looks good!  Only found a minor typo in the `ChangeLog`.

**Pull request merged**